lib/polyfill/patchedmediakeys_20140218.js

```diff
@@ -0,0 +1,832 @@
+
+
+
+/**
+ * An implementation of MediaKeySession.
+ *
+ * @constructor
+ * @param {MSMediaKeys} nativeMediaKeys
+ * @param {string} sessionType
+ * @implements {MediaKeySession}
+ * @extends {shaka.util.FakeEventTarget}
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.
+    MediaKeySession = function(nativeMediaKeys, sessionType) {
+  shaka.log.debug('v20140218.MediaKeySession');
+  shaka.util.FakeEventTarget.call(this, null);
+
+  // Native MediaKeySession, which will be created in generateRequest
+  /** @private @type {MSMediaKeySession} */
+  this.nativeMediaKeySession_ = null;
+
+  /** @private @type {MSMediaKeys} */
+  this.nativeMediaKeys_ = nativeMediaKeys;
+
+  // Promises that are resolved later
+  /** @private @type {Promise} */
+  this.generateRequestPromise_ = null;
+
+  /** @private @type {Promise} */
+  this.updatePromise_ = null;
+
+  /** @private {string} */
+  this.type_ = sessionType;
+
+  /** @type {string} */
+  this.sessionId = '';
+
+  /** @type {number} */
+  this.expiration = NaN;
+
+  /** @type {!shaka.util.PublicPromise} */
+  this.closed = new shaka.util.PublicPromise();
+
+  /** @type {!MediaKeyStatusMap} */
+  this.keyStatuses =
+      new shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeyStatusMap();
+};
+goog.inherits(shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession,
+    shaka.util.FakeEventTarget);
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession.prototype.
+    generateRequest = function(initDataType, initData) {
+  shaka.log.debug('v20140218.MediaKeySession.generateRequest');
+
+  this.generateRequestPromise_ = new shaka.util.PublicPromise();
+
+  try {
+    // This EME spec version requires a MIME content type as the 1st param
+    // to createSession, but doesn't seem to matter what the value is...
+    this.nativeMediaKeySession_ = this.nativeMediaKeys_
+        .createSession('video/mp4', initData);
+
+    // Alias
+    var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+    // Attach session event handlers here
+    this.nativeMediaKeySession_.addEventListener('mskeymessage',
+        /** @type {shaka.util.EventManager.ListenerType} */
+        (this.onMsKeyMessage_.bind(this)), false);
+    this.nativeMediaKeySession_.addEventListener('mskeyadded',
+        /** @type {shaka.util.EventManager.ListenerType} */
+        (this.onMsKeyAdded_.bind(this)), false);
+    this.nativeMediaKeySession_.addEventListener('mskeyerror',
+        /** @type {shaka.util.EventManager.ListenerType} */
+        (this.onMsKeyError_.bind(this)), false);
+
+    this.updateKeyStatus_('status-pending');
+  }
+  catch (exception) {
+    this.generateRequestPromise_.reject(exception);
+  }
+
+  return this.generateRequestPromise_;
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession.prototype.
+    load = function() {
+  shaka.log.debug('v20140218.MediaKeySession.load');
+
+  return Promise.reject(new Error('MediaKeySession.load not yet supported'));
```

Comment:
I believe that because of this, offline playback as designed in Shaka won't work yet.  That's not a deal-breaker, but JFYI.  Since PlayReady does offline in a non-EME-compliant way, getting offline on IE may just have to wait.